src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  ChangeEvent,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from "react"
+import styles from "./SearchInput.module.css"
+
+interface SearchInputProps {
+  ariaLabel: string
+  id?: string
+  name?: string
+  placeholder?: string
+  searchIcon?: ReactNode
+  clearIcon?: ReactNode
+  debounceDelay?: number
+  value?: string
+  onSearch?: (searchTerm: string) => void
+  showSearchButton?: boolean
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  onFocus?: (e: FocusEvent<HTMLInputElement>) => void
+}
+
+export function SearchInput({
+  ariaLabel,
+  id,
+  name,
+  placeholder,
+  searchIcon,
+  clearIcon,
+  onSearch,
+  debounceDelay = 300,
+  showSearchButton,
+  value = "",
+  onBlur,
+  onFocus,
+}: SearchInputProps) {
+  const [searchTerm, setSearchTerm] = useState("")
+  const [cancelButtonHidden, setCancelButtonHidden] = useState(true)
+  const timer = useRef<NodeJS.Timeout>()
+
+  useEffect(() => {
+    setSearchTerm(value)
+  }, [value])
+
+  useEffect(() => {
+    if (searchTerm && !showSearchButton) {
+      timer.current = setTimeout(() => onSearch?.(searchTerm), debounceDelay)
+    }
+    return () => {
+      clearTimeout(timer.current)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think I lead you astray with my previous comment, the suspicious thing here is the:
```
// eslint-disable-next-line react-hooks/exhaustive-deps
```
I can't see why we would need to disable the dependency check and could end up backfiring.